rllib/evaluation/collectors/agent_collector.py

```diff
@@ -552,6 +551,18 @@ def _unflatten_as_buffer_struct(
         """Unflattens the given to match the buffer struct format for that key."""
         if key not in self.buffer_structs:
             return data[0]
+        if key == SampleBatch.ACTIONS and not self.disable_action_flattening:
```

Comment:
Thanks for doing this! Can you explain a little what's happening here for me and Jun?
Have you tested if this breaks other tests /w connectors?

Reply:
action flattening is never actually happening at any point in the episode. This fix enables action flattening to happen